storage/storageCacherAdapter/storageCacherAdapter.go

```diff
@@ -0,0 +1,243 @@
+package storageCacherAdapter
+
+import (
+	"math"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("storageCacherAdapter")
+
+type storageCacherAdapter struct {
+	cacher storage.AdaptedSizedLRUCache
+	db     storage.Persister
+	lock   sync.RWMutex
+
+	storedDataFactory storage.StoredDataFactory
+	marshalizer       marshal.Marshalizer
+}
+
+// NewStorageCacherAdapter creates a new storageCacherAdapter
+func NewStorageCacherAdapter(
+	cacher storage.AdaptedSizedLRUCache,
+	db storage.Persister,
+	storedDataFactory storage.StoredDataFactory,
+	marshalizer marshal.Marshalizer,
+) (*storageCacherAdapter, error) {
+	if check.IfNil(cacher) {
+		return nil, storage.ErrNilCacher
+	}
+	if check.IfNil(db) {
+		return nil, storage.ErrNilPersister
+	}
+	if check.IfNil(marshalizer) {
+		return nil, storage.ErrNilMarshalizer
+	}
+	if check.IfNil(storedDataFactory) {
+		return nil, storage.ErrNilStoredDataFactory
+	}
+
+	return &storageCacherAdapter{
+		cacher:            cacher,
+		db:                db,
+		lock:              sync.RWMutex{},
```

Comment:
i think this can be removed

Reply:
It can, but I prefer to leave it like this so I can see all the parameters explicitly initialized.